client/src/components/QuestionsAndAnswers.jsx

```diff
@@ -40,25 +48,35 @@ export default function QuestionsAndAnswers({ productId, productName }) {
       .catch((err) => console.log('Error during get request on Q/A part for questions and answers list'));
   }, []);
 
+  const [searchText, setSearchText] = useState('');
+  const handleSearch = function (event) {
+    if (event.target.value.length >= 3 || event.target.value.length === 0) {
```

Comment:
"If the user clears the search term, or removes characters so that less than 3 remain, the list should return to the state where it is not filtered to matching text."

I think this might have to be:
```
if event target value length >= 3
  set searchText to event target value
else
  set searchText to ''
```

I think atm, if you have a search longer than 2, and then delete characters so that you have 1 character left, the search term will still stay at the 3 letter term? Maybe I'm interpreting this incorrectly

Reply:
at first, I thought state need to be changed when the chars more than 3 or as an empty string, only within those two situations change state, and render page ...I made the change as you suggested, now when I typed in 3 letters, and as I delete letters, page can be rerended..